WPFHospitalEditor/Building.xaml.cs

```diff
@@ -101,12 +62,12 @@ private void setFloorComboBox()
         private void back_Click(object sender, RoutedEventArgs e)
         {
             clearAll();           
-            HospitalMap hospitalMap = new HospitalMap(mapObjectController.getAllMapObjects());
             this.Close();
-            hospitalMap.Show();
+            CanvasService.addObjectToCanvas(mapObjectController.getOutterMapObjects(mapObjectController.getAllMapObjects()), HospitalMap.canvasHospitalMap);
```

Comment:
Why you use mapObjectController twice in same line

Reply:
I changed getOutterMapObjects method in MapObjectRepository, and now mapObjectController is used only once in this line.